File: src/my_project/easy_problems/from201to250/rings_and_rods.py
from typing import List 

class Solution:
    def countPoints(self, rings: str) -> int:
        
        dict_answer = {}
        
        numbers = '0123456789'
        
        for i in range(10):
            dict_answer[numbers[i]] = []
            
        for i in range(0,len(rings),2):
            
            if rings[i] not in dict_answer[rings[i+1]]:
                dict_answer[rings[i+1]].append(rings[i]) 
            
        answer: int = 0
        
        for i in range(10):
            if len(dict_answer[numbers[i]]) == 3:
                answer += 1
                
        
        
        
        return answer 
    
solution = Solution()

print(solution.countPoints(rings = "B0B6G0R6R0R6G9"))


# function to be reverted
def function_to_be_reverted(nums):
    return nums

File: src/my_project_typescript/easy_problems_typescript/from101to150/ringsAndRods.ts
function countPoints(rings: string): number {

    let objectAnswer:  {[key:string]: Array<string>} = {}

    let numbers: string = '0123456789'


    for (let i = 0; i < 10; i++){
        objectAnswer[numbers[i]] = []
    }

    for (let i = 0; i < rings.length; i = i + 2){
        if (!objectAnswer[rings[i+1]].includes(rings[i])){
            objectAnswer[rings[i+1]].push(rings[i])
        }
    }

    let answer: number = 0

    for (let i = 0; i < 10; i++){
        if (objectAnswer[numbers[i]].length === 3){
            answer += 1
        }
    }

    return answer

};